### PR TITLE
Avoid finding ending key index for bitmap equal query.

### PR DESCRIPTION
**Pull request by lee-lei**
## What changes were proposed in this pull request?
    1. Use binary search to find starting or ending index.
    2. If equal query, no need to find ending index.
    3. Correct starting index or ending index for dummy starting
       key and dummy ending key.
    4. Add one unit test.
## How was this patch tested?
unit tests all passed.

### CODE REVIEW

File: src/main/scala/org/apache/spark/sql/execution/datasources/oap/index/BitMapScanner.scala [hunk 1]
```diff
@@ -159,33 +160,32 @@ private[oap] case class BitMapScanner(idxMeta: IndexMeta) extends IndexScanner(i
...
     } else {
-      // find first key which >= start key, can't find return -1
+      // Below will use binary search, which is more efficient than traversing from the first key.
+      // If no found, return -1.
       if (range.startInclude) {
-        keySeq.indexWhere(ordering.compare(range.start, _) <= 0)
+        keySeq.search(range.start)(ordering).insertionPoint
       } else {
-        keySeq.indexWhere(ordering.compare(range.start, _) < 0)
+        keySeq.search(range.start)(ordering).insertionPoint + 1
```
Comment (linhong-intel) [inline src/main/scala/org/apache/spark/sql/execution/datasources/oap/index/BitMapScanner.scala]: if no found, return -1?
-1 + 1 = 0?

Comment (linhong-intel) [reply src/main/scala/org/apache/spark/sql/execution/datasources/oap/index/BitMapScanner.scala]: Seems the code is not correct and current test can't find this problem.
Can you add some test cases?

**Comment by lee-lei** [comment]
The new version will be uploaded soon. Thanks.

**Comment by lee-lei** [comment]
Done, Please help review when you are available. Thanks. @linhong-intel @adrian-wang @yueawang

**Comment by lee-lei** [comment]
You are right. I will correct and upload it at once. Thanks.

**Pull request merged**